web/src/app/administration/classification/details/classification-details.component.ts

```diff
@@ -224,18 +222,17 @@ export class ClassificationDetailsComponent implements OnInit, OnDestroy {
   }
 
   private async selectClassification(id: string) {
-    if (this.classificationIsAlreadySelected()) {
-      return true;
+    if (!this.classificationIsAlreadySelected()) {
+      this.requestInProgress = true;
+      const classification = await this.classificationsService.getClassification(id);
+      this.fillClassificationInformation(classification);
+      this.classificationsService.selectClassification(classification);
+      this.requestInProgress = false;
     }
-    this.requestInProgress = true;
-    const classification = await this.classificationsService.getClassification(id);
-    this.fillClassificationInformation(classification);
-    this.classificationsService.selectClassification(classification);
-    this.requestInProgress = false;
   }
 
   private classificationIsAlreadySelected(): boolean {
-    if (this.action === ACTION.CREATE && this.classification) { return true; }
+    return this.action === ACTION.CREATE && !!this.classification;
```

Comment:
returns true & false now. intended?

Reply:
It's definitely intended because a function should always return a value fro the same type - in this case boolean.